ambari-server/src/test/java/org/apache/ambari/server/api/services/stackadvisor/StackAdvisorHelperTest.java

```diff
@@ -157,7 +157,7 @@ public void testCreateRecommendationCommand_returnsComponentLayoutRecommendation
     ServiceInfo service = mock(ServiceInfo.class);
     when(metaInfo.getService(Mockito.anyString(), Mockito.anyString(), Mockito.anyString())).thenReturn(service);
     when(service.getServiceAdvisorType()).thenReturn(ServiceInfo.ServiceAdvisorType.PYTHON);
-    StackAdvisorHelper helper = new StackAdvisorHelper(configuration, saRunner, metaInfo);
+    StackAdvisorHelper helper = new StackAdvisorHelper(configuration, saRunner, metaInfo, null);
```

Comment:
Would not be better to have an overloaded constructor? So that you should not pass `null`s all over the places.

Reply:
I rather extracted out the object creation to a new method. I don't want to add an overload because we're only using null-s in the test.